sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -296,6 +310,17 @@ case class DataSourceResolution(
       orCreate = replace.orCreate)
   }
 
+  private def convertDeleteFrom(
+      catalog: TableCatalog,
+      identifier: Identifier,
+      delete: DeleteFromStatement): DeleteFromTable = {
+    val relation = CatalogV2Util.loadTable(catalog, identifier)
+        .map(DataSourceV2Relation.create).getOrElse(UnresolvedRelation(delete.tableName))
+    val aliased = delete.tableAlias.map { SubqueryAlias(_, relation) }.getOrElse(relation)
+    val filter = Filter(delete.condition, aliased)
+    DeleteFromTable(aliased, filter)
```

Comment:
We can simplify it: just create a `DeleteFromTable(aliased, delete.condition)`. The analyzer will help you resolve the expressions in the logical plan.

Reply:
Thanks @cloud-fan . In fact my first try is `DeleteFromTable(aliased, delete.condition)`, however, for case like delete with a subquery, the resolve would throw exception because both the `ResolveReferences` and `ResolveSubqueries` do not consider the case that a subquery in an expression (analyzer can deal with subquery in a filter). To fix this, we have two opts, 1, use `Filter` other than `Expression` in `DeleteFromTable`, or 2, add additional resolve codes in `ResolveReferences` or `ResolveSubqueries`. I adopted the 1 one.
What do you think?